test_parallel/test_fwi.py

```diff
@@ -0,0 +1,142 @@
+    "degree": 4,  # p order
+    "dimension": 2,  # dimension
+}
+dictionary["parallelism"] = {
+    "type": "automatic",  # options: automatic (same number of cores for evey processor) or spatial
+}
+dictionary["mesh"] = {
+    "Lz": 2.0,  # depth in km - always positive   # Como ver isso sem ler a malha?
+    "Lx": 2.0,  # width in km - always positive
+    "Ly": 0.0,  # thickness in km - always positive
+    "mesh_file": None,
+    "mesh_type": "firedrake_mesh",
+}
+dictionary["acquisition"] = {
+    "source_type": "ricker",
+    "source_locations": spyro.create_transect((-0.55, 0.7), (-0.55, 1.3), 6),
+    # "source_locations": [(-1.1, 1.5)],
+    "frequency": 5.0,
+    "delay": 0.2,
+    "delay_type": "time",
+    "receiver_locations": spyro.create_transect((-1.45, 0.7), (-1.45, 1.3), 200),
+}
+dictionary["time_axis"] = {
+    "initial_time": 0.0,  # Initial time for event
+    "final_time": final_time,  # Final time for event
+    "dt": 0.001,  # timestep size
+    "amplitude": 1,  # the Ricker has an amplitude of 1.
+    "output_frequency": 100,  # how frequently to output solution to pvds - Perguntar Daiane ''post_processing_frequnecy'
+    "gradient_sampling_frequency": 1,  # how frequently to save solution to RAM    - Perguntar Daiane 'gradient_sampling_frequency'
+}
+dictionary["visualization"] = {
+    "forward_output": False,
+    "forward_output_filename": "results/forward_output.pvd",
+    "fwi_velocity_model_output": False,
+    "velocity_model_filename": None,
+    "gradient_output": False,
+    "gradient_filename": "results/Gradient.pvd",
+    "adjoint_output": False,
+    "adjoint_filename": None,
+    "debug_output": False,
+}
+dictionary["inversion"] = {
+    "perform_fwi": True,  # switch to true to make a FWI
+    "initial_guess_model_file": None,
+    "shot_record_file": None,
+}
+
+
+def test_fwi(load_real_shot=False, use_rol=False):
+    """
+    Run the Full Waveform Inversion (FWI) test.
+
+    Parameters
+    ----------
+        load_real_shot (bool, optional): Whether to load a real shot record or not. Defaults to False.
+    """
+
+    # Setting up to run synthetic real problem
+    if load_real_shot is False:
+        FWI_obj = spyro.FullWaveformInversion(dictionary=dictionary)
+
+        FWI_obj.set_real_mesh(mesh_parameters={"dx": 0.1})
+        center_z = -1.0
+        center_x = 1.0
+        mesh_z = FWI_obj.mesh_z
+        mesh_x = FWI_obj.mesh_x
+        cond = fire.conditional((mesh_z-center_z)**2 + (mesh_x-center_x)**2 < .2**2, 3.0, 2.5)
+
+        FWI_obj.set_real_velocity_model(conditional=cond, output=True, dg_velocity_model=False)
+        FWI_obj.generate_real_shot_record(
+            plot_model=True,
+            filename="True_experiment.png",
+            abc_points=[(-0.5, 0.5), (-1.5, 0.5), (-1.5, 1.5), (-0.5, 1.5)]
+        )
+        np.save("real_shot_record", FWI_obj.real_shot_record)
+
+        spyro_shots = FWI_obj.real_shot_record
```

Comment:
Is this object used?

Reply:
No, it was leftover code from debugging. Removed it in the latest commit.